src/app/extensions/captcha/shared/captcha-v3/captcha-v3.component.ts

```diff
@@ -41,7 +41,18 @@ export class CaptchaV3Component implements OnInit, OnDestroy {
           take(1),
           switchMap(() =>
             timer(0, 2 * (60 - 3) * 1000).pipe(
-              switchMap(() => this.recaptchaV3Service.execute(this.parentForm.get('captchaAction').value))
+              switchMap(() =>
+                this.recaptchaV3Service.execute(this.parentForm.get('captchaAction').value).pipe(
+                  first(),
+                  tap(token => {
+                    if (!token) {
```

Comment:
use the rxjs filter operator before the tap,instead of the if 
is it intended to test the result only once?

Reply:
It is not necessary to use wether first() as well filter(). Condition will be checked within the tap method.